### PR TITLE
Plat 5040 call new api

### PR DESCRIPTION
**Pull request by gadyaari**

### CODE REVIEW

File: lib/BackendClient.js [hunk 1]
```diff
@@ -128,11 +126,11 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
-    function getEntries(list) {
+    function getEntries(list, logger) {
```
Comment (kalturaguy) [inline lib/BackendClient.js]: not sure on the order of parameters, i think logger should be first?

File: lib/BackendClient.js [hunk 2]
```diff
@@ -204,33 +201,34 @@ var BackendClient = (function(){
...
             });
     };
 
-    BackendClient.getEntries = function(entryList) {
-        return createSession().
-                then(function () {
-                    return getEntries(entryList);
-                });
+    BackendClient.getEntries = function(entryList, logger) {
+        return createSession(logger)
```
Comment (kalturaguy) [inline lib/BackendClient.js]: actully for createSession it doesn't really matter which logger it is, you can put in the general moduel logger

File: lib/entry/FlavorDownloader.js [hunk 3]
```diff
@@ -83,11 +84,11 @@ var FlavorDownloader = (function() {
 
     var downloadIndefinitely = function(onIterationEndCallback) {
         var that = this;
-        that.logger.info("starting iteration");
         if (!that.shouldRun) {
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: why did you delete the starting iteration log?

File: lib/entry/FlavorDownloader.js [hunk 4]
```diff
@@ -83,11 +84,11 @@ var FlavorDownloader = (function() {
...
             downloaderEnd.resolve();
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: please put here a log

File: lib/entry/FlavorDownloader.js [hunk 5]
```diff
@@ -96,9 +97,9 @@ var FlavorDownloader = (function() {
             })
             .catch(function failure(err) {
                 that.emit("iteration-error", err);
-                that.logger.error("iteration error: %s", err);
+                that.logger.error("iteration error: %s", err.message);
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: if there is an exception we will lose the call stack  (util.inspect)

File: lib/entry/LiveEntry.js [hunk 6]
```diff
@@ -103,41 +93,46 @@ LiveEntry.prototype.start = function() {
...
+                    that.logger.error("Failed  to start flavor downloaders for some of the flavors:\n", errs.err);
                 }
             });
 };
 
 LiveEntry.prototype.stop = function() {
     var that = this;
+
+    if (that.FSM.current === 'stopped') {
+        return;
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: put here a log telling that stop is pending

File: lib/utils/fsm.js [hunk 7]
```diff
@@ -123,10 +123,10 @@
                 if (this.cannot(name))
                     return this.error(name, from, to, args, StateMachine.Error.INVALID_TRANSITION, "event " + name + " inappropriate in current state " + this.current);
```
Comment (kalturaguy) [inline lib/utils/fsm.js]: remove remarks

File: lib/BackendClient.js (hunk 1 above)
Comment (gadyaari) [inline lib/BackendClient.js]: If we put logger first it will cause a problem whenever someone is trying to call method in the future without passing a logger as a parameter. I think it's an option we should keep open... never know what's gonna be in the future.

File: lib/entry/LiveEntry.js (hunk 6 above)
Comment (gadyaari) [inline lib/entry/LiveEntry.js]: done
